### PR TITLE
tests: drivers: uart_async_api: fix userspace fault

### LINKED ISSUE #71983: tests: driver: drivers.uart.async_api.lpuart: mimxrt10xx_evk: hardfault
This issue list is only for bugs in the main Zephyr code base
(https://github.com/zephyrproject-rtos/zephyr/). If the bug is for a project
fork (such as NCS) specific feature, please open an issue in the fork project
instead.

**Describe the bug**
tests/drivers/uart/uart_async_api meet hardfault

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 mimxrt1010_evk
 - What have you tried to diagnose or workaround this issue?
 NA
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?

```
a5cf757c9ecbaa562a4f31b58bf133a4b978da8a is the first bad commit
commit a5cf757c9ecbaa562a4f31b58bf133a4b978da8a
Author: Dat Nguyen Duy <dat.nguyenduy@nxp.com>
Date:   Wed Jun 14 09:38:41 2023 +0700

    drivers: dma_mcux_edma: improve interrupt handling

    The current implementation iterates over all channels
    even if only several channels share the same irq. This
    introduces one more dt property to describe an offset
    between two channels share the same interrupt id.

    Beside that, the error interrupt must be put as last
    element of "interrupt" dt property.

    Signed-off-by: Dat Nguyen Duy <dat.nguyenduy@nxp.com>

 drivers/dma/dma_mcux_edma.c         | 105 ++++++++++++++++++------------------
 dts/arm/nxp/nxp_rt10xx.dtsi         |   1 +
 dts/arm/nxp/nxp_rt11xx.dtsi         |   2 +
 dts/bindings/dma/nxp,mcux-edma.yaml |   7 +++
 4 files changed, 61 insertions(+), 54 deletions(-)

```

**To Reproduce**

west twister -p mimxrt1010_evk -T tests/drivers/uart/uart_async_api

**Expected behavior**

PASS

**Impact**

DMA issue iwth lpuart in this platform. (other RT series is ok including RT1015)

**Logs and console output**

```
*** Booting Zephyr OS build v3.6.0-3046-g78150c8e1b2c ***
Running TESTSUITE uart_async_chain_read
===================================================================
E: No buffers to release from RX DMA!
E: No buffers to release from RX DMA!
START - test_chained_read
E: 0x60011a50 used before initialization
E: syscall z_vrfy_uart_rx_enable failed check: access denied
E: r0/a1:  0x00000000  r1/a2:  0x00000000  r2/a3:  0x00000000
E: r3/a4:  0x00000000 r12/ip:  0x00000000 r14/lr:  0x00000000
E:  xpsr:  0x00000000
E: Faulting instruction address (r15/pc): 0x3f750880
E: >>> ZEPHYR FATAL ERROR 3: Kernel oops on CPU 0
E: Current thread: 0x20002840 (test_chained_read)
E: Halting system

```

**Environment (please complete the following information):**
 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.6.0-3046-g78150c8e1b2c

### PR DESCRIPTION
**Pull request by EmilioCBen**
Fixes https://github.com/zephyrproject-rtos/zephyr/issues/71983

### CODE REVIEW

**Review by decsny — DISMISSED** [review]
pay attention to what I did in this PR https://github.com/zephyrproject-rtos/zephyr/pull/71900/files

**Comment by EmilioCBen** [comment]
Greetings @dcpleung and @andrewboie , I was wondering if you could take a look at this PR?
Context [This PR](https://github.com/zephyrproject-rtos/zephyr/pull/70503) added the ability to use the nocache region on the uart_async_api test. This in return broke some nxp platforms, for instance here with the mimxrt1064_evk, when using the sdram0 for sram, the 1064 can take advantage of using the nocache memory, but Userspace is unable to access the nocache region at the moment. What are your thoughts, should we be using Userspace for the uart test? I also notice @andrewboie, you explicitly set Userspace to be enabled, I know this may have been some time ago, but is there a particular reason you explicitly enabled it?
Happy to hear feedback from both of you.

**Comment by dcpleung** [comment]
Which test fails? There is a test `drivers.uart.async_api.nocache_mem` which explicitly disables userspace when there is nocache region.

**Comment by decsny** [comment]
commit message is out of date

File: tests/drivers/uart/uart_async_api/testcase.yaml [hunk 1]
```diff
@@ -81,9 +81,6 @@ tests:
     harness_config:
       fixture: gpio_loopback
     depends_on: gpio
-    platform_allow:
-      - nucleo_f746zg
-      - nucleo_h723zg
```
Comment (hakehuang) [inline tests/drivers/uart/uart_async_api/testcase.yaml:86]: better not change this but add supported platform instread

Comment (EmilioCBen) [reply tests/drivers/uart/uart_async_api/testcase.yaml:86]: I was trying to see why this was added, what platforms are failing if this is not here, but also I was going to confirm if this part of the yaml file is even working, when I build with these two platforms, the extra Kconfig did not appear, neither did they appear for any of the rt series which is causing the tests to fail, however if I manually add the extra Kconfigs on the command line, the rt series passes.
@dcpleung @danieldegrasse 
What am I missing from the file with this information?

**Comment by EmilioCBen** [comment]
Added DNM tag, platforms are not passing right now

**Comment by hakehuang** [comment]
@EmilioCBen , now all board testing PASS with CONFIG_USERSPACE=n for mcux series, but there is a annoying message for this case, for uart driver.
I do not understand the code below in the uart_mcux_lpuart.c function mcux_lpuart_rx_disable
```
	/* No active RX buffer, cannot disable */
	if (!data->async.rx_dma_params.buf) {
		LOG_ERR("No buffers to release from RX DMA!");
``` 
why send thie error message
```
2024-09-25 03:03:34,938 - twister - DEBUG - DEVICE: *** Booting Zephyr OS build v3.7.0-3110-g53dbab9a2da5 ***
2024-09-25 03:03:34,942 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_chain_read
2024-09-25 03:03:34,948 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:34,951 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:34,955 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:34,957 - twister - DEBUG - DEVICE: START - test_chained_read
2024-09-25 03:03:35,386 - twister - DEBUG - DEVICE: PASS - test_chained_read in 0.426 seconds
2024-09-25 03:03:35,392 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:35,396 - twister - DEBUG - DEVICE: TESTSUITE uart_async_chain_read succeeded
2024-09-25 03:03:35,400 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_chain_write
2024-09-25 03:03:35,406 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:35,409 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:35,412 - twister - DEBUG - DEVICE: START - test_chained_write
2024-09-25 03:03:35,417 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:35,420 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:35,424 - twister - DEBUG - DEVICE: PASS - test_chained_write in 0.009 seconds
2024-09-25 03:03:35,430 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:35,434 - twister - DEBUG - DEVICE: TESTSUITE uart_async_chain_write succeeded
2024-09-25 03:03:35,437 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_double_buf
2024-09-25 03:03:35,443 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:35,447 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:35,449 - twister - DEBUG - DEVICE: START - test_double_buffer
2024-09-25 03:03:36,112 - twister - DEBUG - -- west flash: using runner linkserver
-- runners.linkserver: LinkServer: /usr/local/LinkServer/LinkServer, version v1.6.121

2024-09-25 03:03:38,855 - twister - DEBUG - DEVICE: PASS - test_double_buffer in 3.403 seconds
2024-09-25 03:03:38,861 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:38,865 - twister - DEBUG - DEVICE: TESTSUITE uart_async_double_buf succeeded
2024-09-25 03:03:38,868 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_long_buf
2024-09-25 03:03:38,874 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:38,878 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:38,880 - twister - DEBUG - DEVICE: START - test_long_buffers
2024-09-25 03:03:39,045 - twister - DEBUG - DEVICE: PASS - test_long_buffers in 0.162 seconds
2024-09-25 03:03:39,051 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:39,055 - twister - DEBUG - DEVICE: TESTSUITE uart_async_long_buf succeeded
2024-09-25 03:03:39,058 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_multi_rx
2024-09-25 03:03:39,064 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:39,067 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:39,070 - twister - DEBUG - DEVICE: START - test_multiple_rx_enable
2024-09-25 03:03:39,474 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:39,678 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:39,782 - twister - DEBUG - DEVICE: PASS - test_multiple_rx_enable in 0.709 seconds
2024-09-25 03:03:39,789 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:39,792 - twister - DEBUG - DEVICE: TESTSUITE uart_async_multi_rx succeeded
2024-09-25 03:03:39,795 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_read_abort
2024-09-25 03:03:39,802 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:39,805 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:39,807 - twister - DEBUG - DEVICE: START - test_read_abort
2024-09-25 03:03:40,930 - twister - DEBUG - DEVICE: PASS - test_read_abort in 1.120 seconds
2024-09-25 03:03:40,936 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:40,940 - twister - DEBUG - DEVICE: TESTSUITE uart_async_read_abort succeeded
2024-09-25 03:03:40,944 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_single_read
2024-09-25 03:03:40,949 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:40,953 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:40,955 - twister - DEBUG - DEVICE: START - test_single_read
2024-09-25 03:03:41,310 - twister - DEBUG - DEVICE: PASS - test_single_read in 0.352 seconds
2024-09-25 03:03:41,316 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:41,320 - twister - DEBUG - DEVICE: TESTSUITE uart_async_single_read succeeded
2024-09-25 03:03:41,323 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_timeout
2024-09-25 03:03:41,329 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:41,332 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:41,335 - twister - DEBUG - DEVICE: START - test_forever_timeout
2024-09-25 03:03:43,347 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:44,339 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:44,343 - twister - DEBUG - DEVICE: PASS - test_forever_timeout in 3.004 seconds
2024-09-25 03:03:44,349 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:44,352 - twister - DEBUG - DEVICE: TESTSUITE uart_async_timeout succeeded
2024-09-25 03:03:44,356 - twister - DEBUG - DEVICE: Running TESTSUITE uart_async_write_abort
2024-09-25 03:03:44,362 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:44,365 - twister - DEBUG - DEVICE: E: No buffers to release from RX DMA!
2024-09-25 03:03:44,368 - twister - DEBUG - DEVICE: START - test_write_abort
2024-09-25 03:03:44,472 - twister - DEBUG - DEVICE: PASS - test_write_abort in 0.102 seconds
2024-09-25 03:03:44,478 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:44,482 - twister - DEBUG - DEVICE: TESTSUITE uart_async_write_abort succeeded
2024-09-25 03:03:44,486 - twister - DEBUG - DEVICE: ------ TESTSUITE SUMMARY START ------
2024-09-25 03:03:44,496 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_chain_read]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.426 seconds
2024-09-25 03:03:44,502 - twister - DEBUG - DEVICE: - PASS - [uart_async_chain_read.test_chained_read] duration = 0.426 seconds
2024-09-25 03:03:44,512 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_chain_write]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.009 seconds
2024-09-25 03:03:44,519 - twister - DEBUG - DEVICE: - PASS - [uart_async_chain_write.test_chained_write] duration = 0.009 seconds
2024-09-25 03:03:44,529 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_double_buf]: pass = 1, fail = 0, skip = 0, total = 1 duration = 3.403 seconds
2024-09-25 03:03:44,536 - twister - DEBUG - DEVICE: - PASS - [uart_async_double_buf.test_double_buffer] duration = 3.403 seconds
2024-09-25 03:03:44,546 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_long_buf]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.162 seconds
2024-09-25 03:03:44,552 - twister - DEBUG - DEVICE: - PASS - [uart_async_long_buf.test_long_buffers] duration = 0.162 seconds
2024-09-25 03:03:44,562 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_multi_rx]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.709 seconds
2024-09-25 03:03:44,569 - twister - DEBUG - DEVICE: - PASS - [uart_async_multi_rx.test_multiple_rx_enable] duration = 0.709 seconds
2024-09-25 03:03:44,579 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_read_abort]: pass = 1, fail = 0, skip = 0, total = 1 duration = 1.120 seconds
2024-09-25 03:03:44,586 - twister - DEBUG - DEVICE: - PASS - [uart_async_read_abort.test_read_abort] duration = 1.120 seconds
2024-09-25 03:03:44,596 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_single_read]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.352 seconds
2024-09-25 03:03:44,603 - twister - DEBUG - DEVICE: - PASS - [uart_async_single_read.test_single_read] duration = 0.352 seconds
2024-09-25 03:03:44,612 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_timeout]: pass = 1, fail = 0, skip = 0, total = 1 duration = 3.004 seconds
2024-09-25 03:03:44,619 - twister - DEBUG - DEVICE: - PASS - [uart_async_timeout.test_forever_timeout] duration = 3.004 seconds
2024-09-25 03:03:44,629 - twister - DEBUG - DEVICE: SUITE PASS - 100.00% [uart_async_write_abort]: pass = 1, fail = 0, skip = 0, total = 1 duration = 0.102 seconds
2024-09-25 03:03:44,636 - twister - DEBUG - DEVICE: - PASS - [uart_async_write_abort.test_write_abort] duration = 0.102 seconds
2024-09-25 03:03:44,639 - twister - DEBUG - DEVICE: ------ TESTSUITE SUMMARY END ------
2024-09-25 03:03:44,645 - twister - DEBUG - DEVICE: ===================================================================
2024-09-25 03:03:44,649 - twister - DEBUG - DEVICE: RunID: 02c9747a5bbfb924d0ba261679e00679
2024-09-25 03:03:44,651 - twister - DEBUG - DEVICE: PROJECT EXECUTION SUCCESSFUL
```
this is not related to this fixing PR.  maybe we can work this with a new PR. Thanks

**Review by hakehuang — DISMISSED** [review]

**Comment by EmilioCBen** [comment]
Thank you @hakehuang, I can look into this error message!

**Review by decsny — DISMISSED** [review]

File: tests/drivers/uart/uart_async_api/testcase.yaml [hunk 2]
```diff
@@ -61,6 +61,10 @@ tests:
     filter: CONFIG_SERIAL_SUPPORT_ASYNC and CONFIG_UART_MCUX_LPUART
     harness: ztest
     depends_on: dma
+    extra_configs:
```
Comment (danieldegrasse) [inline tests/drivers/uart/uart_async_api/testcase.yaml:64]: Will this work? Some platforms (like the K66 or K8X) do not have support for NOCACHE memory (no ARMv7 MPU), but do use the LPUART

**Review by danieldegrasse — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

**Pull request merged**